code/search/SearchIndex.php

```diff
@@ -77,26 +98,49 @@ public function fieldData($field, $forceType = null, $extraOptions = array())
             foreach ($lookups as $lookup) {
                 $next = array();
 
-                foreach ($sources as $source => $options) {
-                    $class = null;
+                foreach ($sources as $source => $baseOptions) {
+                    $source = $this->getSourceName($source);
 
-                    foreach (SearchIntrospection::hierarchy($source, $options['include_children']) as $dataclass) {
+                    foreach (SearchIntrospection::hierarchy($source, $baseOptions['include_children']) as $dataclass) {
+                        $class = null;
+                        $options = $baseOptions;
                         $singleton = singleton($dataclass);
 
                         if ($hasOne = $singleton->has_one($lookup)) {
+                            // we only want to include base class for relation, omit classes that inherited the relation
```

Comment:
I don't think there is a functional difference between adding the inherited relations or not as the index will fall back on the parent class relation. The difference would be the content of the XML which is output of this process. We probably don't want to have the XML full of fields that have no impact on the search.

Of course I'm happy to discuss this further as I'm quite new to this module.